app/gui2/ydoc-server/languageServerSession.ts

```diff
@@ -466,13 +471,21 @@ class ModulePersistence extends ObservableV2<{ removed: () => void }> {
     if (this.syncedContent == null || this.syncedVersion == null) return
 
     const code = newCode ?? synced.code
-    const newMetadataJson =
-      newMetadata &&
-      json.stringify({ ...this.syncedMeta, ide: { ...this.syncedMeta.ide, node: newMetadata } })
-    const newIdMapJson = newIdMap && serializeIdMap(newIdMap)
+    const metadataToPersist = {
+      ...this.syncedMeta,
+      ide: { ...this.syncedMeta.ide, node: newMetadata },
+    }
+    const newMetadataJson = newMetadata && json.stringify(metadataToPersist)
```

Comment:
I would keep `metadataToPersist` inside `json.stringify` here, since the variable is not actually correct to use outside of the context when metadata is being updated.